src/module/actor/helpers.ts

```diff
@@ -252,6 +253,55 @@ function createEncounterRollOptions(actor: ActorPF2e): Record<string, boolean> {
     return Object.fromEntries(entries);
 }
 
+/** Create roll options pertaining to the terrain  the actor is currently in */
+function createTerrainRollOptions(actor: ActorPF2e): Record<string, boolean> {
+    const toAdd = new Set<string>();
+    const sceneTerrainTypes = canvas.scene?.flags.pf2e.terrainTypes ?? [];
+    for (const terrain of sceneTerrainTypes) {
+        toAdd.add(terrain);
+    }
+    const token = actor.getActiveTokens(false, true).at(0);
+    const terrains = ((): Set<string> => {
+        // No token on the scene means no terrain roll options
+        if (!token) return new Set<string>();
+        const toRemove = new Set<string>();
+        for (const region of token.regions ?? []) {
+            if (token.elevation < Number(region.elevation.bottom) || token.elevation > Number(region.elevation.top)) {
+                continue;
+            }
+            for (const behavior of region.behaviors) {
```

Comment:
I think you could remove one layer of identation if you did const pf2eTerrainBehaviors = region.behaviors.filter((b) => b.type === "pf2eTerrain") and then looped on that.